BDArmory/UI/BDAEditorArmorWindow.cs

```diff
@@ -430,52 +454,34 @@ void WindowArmor(int windowID)
-                    showHullMenu, StringUtils.Localize("#LOC_BDArmory_Armor_HullMat"), showHullMenu ? BDArmorySetup.BDGuiSkin.box : BDArmorySetup.BDGuiSkin.button);
-                HullLines += 1.15f;
-
-                if (showHullMenu)
+                if (!hullslist)
                 {
-                    if (isSteel != (isSteel = GUI.Toggle(new Rect(10, (line + armorLines + StatLines + HullLines) * lineHeight, 280, lineHeight), isSteel, StringUtils.Localize("#LOC_BDArmory_Steel"), isSteel ? BDArmorySetup.BDGuiSkin.box : BDArmorySetup.BDGuiSkin.button)))
-                    {
-                        if (isSteel)
-                        {
-                            isWood = false;
-                            isAluminium = false;
-                            hullmat = 3;
-                            CalculateArmorMass(true);
-                        }
-                    }
-                    HullLines += 1.15f;
-                    if (isWood != (isWood = GUI.Toggle(new Rect(10, (line + armorLines + StatLines + HullLines) * lineHeight, 280, lineHeight), isWood, StringUtils.Localize("#LOC_BDArmory_Wood"), isWood ? BDArmorySetup.BDGuiSkin.box : BDArmorySetup.BDGuiSkin.button)))
-                    {
-                        if (isWood)
-                        {
-                            isAluminium = false;
-                            isSteel = false;
-                            hullmat = 1;
-                            CalculateArmorMass(true);
-                        }
-                    }
-                    HullLines += 1.15f;
-                    if (isAluminium != (isAluminium = GUI.Toggle(new Rect(10, (line + armorLines + StatLines + HullLines) * lineHeight, 280, lineHeight), isAluminium, StringUtils.Localize("#LOC_BDArmory_Aluminium"), isAluminium ? BDArmorySetup.BDGuiSkin.box : BDArmorySetup.BDGuiSkin.button)))
-                    {
-                        if (isAluminium)
-                        {
-                            isWood = false;
-                            isSteel = false;
-                            hullmat = 2;
-                            CalculateArmorMass(true);
-                        }
-                    }
-                    HullLines += 1.15f;
-                    if (!isSteel && !isWood && !isAluminium)
+                    FillHullList();
+                    hullslist = true;
+                }
+                GUIStyle listStyle = new GUIStyle(BDArmorySetup.BDGuiSkin.button);
+                listStyle.fixedHeight = 18; //make list contents slightly smaller
+                if (armorLines + StatLines != oldLines)
+                {
+                    hullBox = new BDGUIComboBox(new Rect(10, (line + armorLines + StatLines) * lineHeight, 280, lineHeight), new Rect(10, (line + armorLines + StatLines) * lineHeight, 280, lineHeight), hullBoxText, hullGUI, 60, listStyle);
```

Comment:
As mentioned in #tea, this is showing an empty button initially for some reason despite having nearly identical code to the armour combobox.

Reply:
Hmm. Adding a ``Debug.Log("[HULLDEBUG] HullType string localized to: " + StringUtils.Localize("#LOC_BDArmory_Armor_HullType"));`` line immediately prior to ``hullBoxText.text = StringUtils.Localize("#LOC_BDArmory_Armor_HullType");`` has the debug log print out "HullType string localized to: ", and the materials button properly labeled "Hull Material"
Doublechecking by commenting out the ``GUI.Label(new Rect(40, line * lineHeight, 300, lineHeight), StringUtils.Localize("#LOC_BDArmory_ArmorSelect"), style);`` line above ``if(!armorList) has the select armor material button also labelless.
Looks like the GUIContent.text requires the string already loaded into memory.

Reply:
I see the bug.
```csharp
localizedStrings[template] = Localizer.Format(template);
```
in `StringUtils.Localize` should be
```csharp
result = Localizer.Format(template);
localizedStrings[template] = result;
```
The first time it's accessed it's returning an empty string, but every time after that it's returning the correct one. The above change should fix that.